src/Model/Resolver/MoveWishlistToCart.php

```diff
@@ -95,29 +96,56 @@ public function __construct(
      */
     protected function addItemsToCart(array $wishlistItems, CartInterface $quote): void
     {
-        foreach ($wishlistItems as $item) {
-            $product = $item['product'];
+        $cartItems = $quote->getItems();
+        $addedQty = false;
+        $shouldShowError = false;
+
+        foreach ($cartItems as $item) {
+            $product = $item->getProduct();
+            $qty = $item->getQty();
+            $sku = $product->getSku();
+
+            if (array_key_exists($sku, $wishlistItems)) {
+                $wishlistItem = $wishlistItems[$sku];
+                unset($wishlistItems[$sku]);
 
-            $stockStatus = $this->stockStatusRepository->get($product->getId());
-            $productStockStatus = $stockStatus->getStockStatus();
+                $qty += $wishlistItem['qty'];
+                $item->setQty($qty);
 
-            // Don't add out of stock product to cart
-            if($productStockStatus === StockStatusInterface::STATUS_OUT_OF_STOCK){
-                continue;
+                $wishlistItem['item']->delete();
+                $addedQty = true;
             }
+        }
 
-            $data = json_decode($item['buy_request'], true);
+        $allItemsStatus = false;
+
+        foreach ($wishlistItems as $item) {
+            $product = $item['product'];
+            $data = [];
+
+            if ($product->getTypeId() === Configurable::TYPE_CODE) {
+                $data['super_attribute'] = $item['super_attribute'];
+            }
 
             $buyRequest = new DataObject();
             $buyRequest->setData($data);
```

Comment:
What is this for? Maybe it's possible to get rid of this? Seems like you pass an empty array there, why is it necessary?

Reply:
This seems to have been here before my changes.